### DISCUSSION TITLE
How to visualise/explore InstrumentationLibrary Name and Version on DataDog

### DISCUSSION THREAD

**Discussion by ochauhan** [open-telemetry/opentelemetry-collector-contrib#4960] [Q&A] [answered]
I have set the InstrumentationLibrary Name and Version.

I am able to see these data in the logging exporter as below.

2021-08-30T15:44:40.323+0530	INFO	loggingexporter/logging_exporter.go:56	MetricsExporter	{"#metrics": 1942}
2021-08-30T15:44:40.338+0530	DEBUG	loggingexporter/logging_exporter.go:66	ResourceMetrics #0
InstrumentationLibraryMetrics #0
_**InstrumentationLibrary virtual_stats 14.1.0.6**_
Metric #0
Descriptor:
     -> Name: virtual.clientside.maxConns
     -> Description: 
     -> Unit: 
     -> DataType: Gauge
NumberDataPoints #0
StartTimestamp: 2021-08-30 10:14:39.384255 +0000 UTC
Timestamp: 2021-08-30 10:14:39.384255 +0000 UTC
Value: 0.000000
Metric #1
..
..

In the above highlighted,  Name: virtual_stats and Version: 14.1.0.6
How do I get the InstrumentationLibrary Name and Version in the DataDog?

**Comment by mx-psi** [comment]
Hi @ochauhan! You can enable the `resource_attributes_as_tags` setting to to transform all resource attributes into metric labels, which are then converted into Datadog tags:

```yaml
datadog:
  api:
    key: <api key>
  metrics:
    resource_attributes_as_tags: true
```

In your metric you would get a `virtual.clientside.maxConns` metric tagged by all resource attributes including the instrumentation library name and version. Let me know if that works for you!

**Comment by ochauhan** [comment]
Thanks @mx-psi for your response.
After setting the flag   `resource_attributes_as_tags  to true`, I am able to see the attributes as tags in the Datadog

E.g: resource.Attributes().Insert("endpoint", value)

This "endpoint" attribute is visible in the Datadog.

However, InstrumentationLibrary name and version are not visible in datadog which are set as mentined below:

E.g:
instrumentationLibraryMetrics.InstrumentationLibrary().SetName("virtual_stats")
instrumentationLibraryMetrics.InstrumentationLibrary().SetVersion(version)

How do I see "virtual_stats" and "version" in the Datadog?
Do we need to enable/set any other flag?

**Comment by mx-psi** [comment, marked as answer]
Hey @ochauhan, apologies for the delay. We will be working on this to add the feature to our exporter since it is not currently possible to add it.
